src/builtins/Transformers.jl

```diff
@@ -343,30 +343,50 @@ otherwise, ignore specified features.
     │ 3   │ 1.14708   │ 3     │
 
 """
-@with_kw_noshow mutable struct Standardizer <: Unsupervised
-    features::Vector{Symbol} = Symbol[] # features to be standardized; empty means all
-    ignore::Vector{Symbol} = Symbol[] # features to be ignored
+mutable struct Standardizer <: Unsupervised
+    features::Vector{Symbol} # features to be standardized; empty means all
+    ignore::Bool # features to be ignored
+    ordered_factor::Bool
+    count::Bool
+end
+
+function MLJBase.clean!(transformer::Standardizer)
+    err = ""
+    if isempty(transformer.features) && transformer.ignore
+        err *= "Features to be ignored must be specified in features field."
+    end
+    return err
+end
+
+# keyword constructor
+function Standardizer(
+    ; features=Symbol[], ignore=false, ordered_factor=false, count=false)
+    transformer = Standardizer(features, ignore, ordered_factor, count)
+    message = MLJBase.clean!(transformer)
+    isempty(message) || throw(ArgumentError(message))
+    return transformer
 end
 
 function MLJBase.fit(transformer::Standardizer, verbosity::Int, X)
     all_features = Tables.schema(X).names
     mach_types   = collect(eltype(selectcols(X, c)) for c in all_features)
+    type_tup = [
```

Comment:
Okay, this is old code can be improved with scitype methods and schema. Let's just remove references to machine types which makes the code harder to read. Please do `scitypes = schema(X).scitypes` and remove `mach_types` altogether.

Instead of `type_up` business, can I suggest

```julia
scitypes = [Continuous]
transformer.ordered_factor && push!(scitypes, OrderedFactor)
transformer.count && push!(scitypes, Count)
AllowedScitype = Union{scitype...}
```

And then, see below...

Reply:
Done! Anyway, quick question - how can I generate `OrderedFactor`? I need it for the test.

Thanks! :)

Reply:
Good question:

`v = categorical(rand("kajldjf", 3), ordered=true)`  will do it. Or  `coerce(rand(Char, 3), OrderedFactor)`